test/benchmark/BenchmarkBigIT.java

```diff
@@ -0,0 +1,345 @@
+                Util.timeQuery(subbedQueryString, tx, "first argument bound");
+                Util.timeQuery(subbedQueryString2, tx, "second argument bound");
+                Util.timeQuery(limitedQueryString, tx, "limit " + limit);
+            }
+        }
+    }
+
+    /**
+     * Scalable multi-join test defined as a non-recursive tree of binary joins,
+     * which is expressed using the following inference rules:
+     * <p>
+     * a(X,Y)  :- b1(X,Z), b2(Z,Y).
+     * b1(X,Y) :- c1(X,Z), c2(Z,Y).
+     * b2(X,Y) :- c3(X,Z), c4(Z,Y).
+     * c1(X,Y) :- d1(X,Z), d2(Z,Y).
+     * <p>
+     * The base relations, c2, c3, c4, d1 and d2 are randomly generated
+     * with 1/5 * N instances (reaching a total of N instances) each defined over 1/5 * N entities.
+     * The query is based on the final derived predicate.
+     */
+    @Test
+    public void testMultiJoin() {
+        final int N = 100;
+        final int limit = 100;
+        System.out.println(new Object() {
+        }.getClass().getEnclosingMethod().getName());
+        loadJoinData(N);
+
+        try (TypeDB.Session session = dataSession()) {
+            try (TypeDB.Transaction tx = session.transaction(Arguments.Transaction.Type.READ, new Options.Transaction().infer(true))) {
+                Thing entityId = tx.concepts().getEntityType("genericEntity").getInstances().next();
+                String queryPattern = "(fromRole: $x, toRole: $y) isa A;";
+                String queryString = "match " + queryPattern;
+                String subbedQueryString = "match " +
+                        queryPattern +
+                        " $x iid " + entityId.getIID().toHexString() + ";";
+                String subbedQueryString2 = "match " +
+                        queryPattern +
+                        " $y iid " + entityId.getIID().toHexString() + ";";
+                String limitedQueryString = "match " + queryPattern +
+                        " limit " + limit + ";";
+
+                Util.timeQuery(queryString, tx, "full");
+                Util.timeQuery(subbedQueryString, tx, "first argument bound");
+                Util.timeQuery(subbedQueryString2, tx, "second argument bound");
+                Util.timeQuery(limitedQueryString, tx, "limit " + limit);
+            }
+        }
+    }
+
+    /**
+     * Scalability test defined in terms of number of rules in the system. Creates a simple rule chain based on join operation on two relations:
+     * <p>
+     * R_i(x, y) := R_1(x, z), R_{i-1}(z, y);    i e [2, N],
+     * <p>
+     * initialised with a chain of N x base relation R_1instances.
+     * <p>
+     * The rules are defined in such a way that relation R_j spans j hops of base relations and has (N + 1 - j) instances.
+     * Consequently R_N has a single instance linking the first and last entities in the base relation chain.
+     */
+    @Test
+    public void testJoinRuleChain() {
+        final int N = 200;  // Stack-size of 200 causes stack-overflow
```

Comment:
eh?? That's what is current set ;)

Reply:
Yes. But it's not being run in CI
The quadratic transitivity tests time out as well.

So I'm leaving the benchmark untouched so what fails fails.